server/core/wordio/index.js

```diff
@@ -0,0 +1,134 @@
+const Logger = require('./../logger.js');
+
+/**
+ * Wordio is the thing where you guess words and stuff
+ *
+ * @type {Wordio}
+ */
+class Wordio {
+	/**
+	 * @param {ClientManager} clientManager for attaching to the clients events
+	 * @param {Chat} chat
+	 * @param {Config} config
+	 * @param {IDictionary} dictionary
+	 */
+	constructor(clientManager, chat, config, dictionary) {
+		if (config.get('wordio', 'true') !== 'true') {
+			Logger.info('Wordio is disabled in config, not initializing');
+			return;
+		}
+
+		this.chat = chat;
+		this.dictionary = dictionary;
+		this.clients = {};
+
+		this.maxGuesses = config.get('maxGuesses', 8);
+		this.wordSize = config.get('wordSize', 6);
+		this.currentWord = dictionary.getDailyWord();
+
+		this.identifier = ''+Date.now()+Math.random();
+
+		clientManager.on('new-client', client => this.attachClient(client));
+	}
+
+	/**
+	 * Subscribes a new client to kiro's events
+	 * will send and receive client positions
+	 *
+	 * @param {Client} client
+	 */
+	attachClient(client) {
+		this.clients[client.id] = client;
+		client.wordio_state = [];
+
+		//Bind events to this user:
+		client.on('wordio_guess', data => {
+			this.guess(client, data);
+		});
+		client.on('wordio_sync_state', states => {
+			this.syncState(client, states);
+		});
+
+		client.on('disconnect', () => {
+			delete this.clients[client.id];
+		});
+		client.emit('wordio', this.identifier);
+	}
+
+	getStateForWord(word) {
+		const letters = word.split('');
+		const wordCheck = this.currentWord.split('');
+		// populate initial state to gray for all letters
+		const states = letters.map((el) => {
+			return {
+				letter: el,
+				state: 'gray'
+			}
+		});
+
+		// check for exact/green matches
+		for (let index in letters) {
+			const letter = letters[index];
+
+			if(letter === this.currentWord[index]) {
+				delete wordCheck[index]; // remove found letter
+				states[index].state = 'green';
+			}
+		}
+
+		// check for non-exact/yellow matches
+		for (let index in letters) {
+			const letter = letters[index];
+
+			const letterLocation = wordCheck.indexOf(letter);
+			if (letterLocation !== -1) {
+				delete wordCheck[letterLocation]; // remove found letter
+				states[index].state = 'yellow';
```

Comment:
Another case that should be fixed:

![image](https://user-images.githubusercontent.com/6020242/157909989-ff82720d-c43f-4830-b951-93c629450bf3.png)

the O should be green, not yellow, whenever there's two letters with one match and first one is in the correct position.